src/openvr/__init__.py

```diff
@@ -1,6947 +1,6949 @@
-VRTrackedCameraError_InvalidSharedTextureHandle = ENUM_VALUE_TYPE(110)
-VRTrackedCameraError_FailedToGetGLTextureId = ENUM_VALUE_TYPE(111)
-VRTrackedCameraError_SharedTextureFailure = ENUM_VALUE_TYPE(112)
-VRTrackedCameraError_NoFrameAvailable = ENUM_VALUE_TYPE(113)
-VRTrackedCameraError_InvalidArgument = ENUM_VALUE_TYPE(114)
-VRTrackedCameraError_InvalidFrameBufferSize = ENUM_VALUE_TYPE(115)
-
-EVRTrackedCameraFrameLayout = ENUM_TYPE
-EVRTrackedCameraFrameLayout_Mono = ENUM_VALUE_TYPE(1)
-EVRTrackedCameraFrameLayout_Stereo = ENUM_VALUE_TYPE(2)
-EVRTrackedCameraFrameLayout_VerticalLayout = ENUM_VALUE_TYPE(16)
-EVRTrackedCameraFrameLayout_HorizontalLayout = ENUM_VALUE_TYPE(32)
-
-EVRTrackedCameraFrameType = ENUM_TYPE
-VRTrackedCameraFrameType_Distorted = ENUM_VALUE_TYPE(0)
-VRTrackedCameraFrameType_Undistorted = ENUM_VALUE_TYPE(1)
-VRTrackedCameraFrameType_MaximumUndistorted = ENUM_VALUE_TYPE(2)
-MAX_CAMERA_FRAME_TYPES = ENUM_VALUE_TYPE(3)
-
-EVRDistortionFunctionType = ENUM_TYPE
-VRDistortionFunctionType_None = ENUM_VALUE_TYPE(0)
-VRDistortionFunctionType_FTheta = ENUM_VALUE_TYPE(1)
-VRDistortionFunctionType_Extended_FTheta = ENUM_VALUE_TYPE(2)
-MAX_DISTORTION_FUNCTION_TYPES = ENUM_VALUE_TYPE(3)
-
-EVSync = ENUM_TYPE
-VSync_None = ENUM_VALUE_TYPE(0)
-VSync_WaitRender = ENUM_VALUE_TYPE(1)
-VSync_NoWaitRender = ENUM_VALUE_TYPE(2)
-
-EVRMuraCorrectionMode = ENUM_TYPE
-EVRMuraCorrectionMode_Default = ENUM_VALUE_TYPE(0)
-EVRMuraCorrectionMode_NoCorrection = ENUM_VALUE_TYPE(1)
-
-Imu_OffScaleFlags = ENUM_TYPE
-OffScale_AccelX = ENUM_VALUE_TYPE(1)
-OffScale_AccelY = ENUM_VALUE_TYPE(2)
-OffScale_AccelZ = ENUM_VALUE_TYPE(4)
-OffScale_GyroX = ENUM_VALUE_TYPE(8)
-OffScale_GyroY = ENUM_VALUE_TYPE(16)
-OffScale_GyroZ = ENUM_VALUE_TYPE(32)
-
-EVRApplicationError = ENUM_TYPE
-VRApplicationError_None = ENUM_VALUE_TYPE(0)
-VRApplicationError_AppKeyAlreadyExists = ENUM_VALUE_TYPE(100)
-VRApplicationError_NoManifest = ENUM_VALUE_TYPE(101)
-VRApplicationError_NoApplication = ENUM_VALUE_TYPE(102)
-VRApplicationError_InvalidIndex = ENUM_VALUE_TYPE(103)
-VRApplicationError_UnknownApplication = ENUM_VALUE_TYPE(104)
-VRApplicationError_IPCFailed = ENUM_VALUE_TYPE(105)
-VRApplicationError_ApplicationAlreadyRunning = ENUM_VALUE_TYPE(106)
-VRApplicationError_InvalidManifest = ENUM_VALUE_TYPE(107)
-VRApplicationError_InvalidApplication = ENUM_VALUE_TYPE(108)
-VRApplicationError_LaunchFailed = ENUM_VALUE_TYPE(109)
-VRApplicationError_ApplicationAlreadyStarting = ENUM_VALUE_TYPE(110)
-VRApplicationError_LaunchInProgress = ENUM_VALUE_TYPE(111)
-VRApplicationError_OldApplicationQuitting = ENUM_VALUE_TYPE(112)
-VRApplicationError_TransitionAborted = ENUM_VALUE_TYPE(113)
-VRApplicationError_IsTemplate = ENUM_VALUE_TYPE(114)
-VRApplicationError_SteamVRIsExiting = ENUM_VALUE_TYPE(115)
-VRApplicationError_BufferTooSmall = ENUM_VALUE_TYPE(200)
-VRApplicationError_PropertyNotSet = ENUM_VALUE_TYPE(201)
-VRApplicationError_UnknownProperty = ENUM_VALUE_TY
```

Comment:
This one is another non-conforming case:
```c++
virtual EVRApplicationError GetStartingApplication( VR_OUT_STRING() char *pchAppKeyBuffer, uint32_t unAppKeyBufferLen ) = 0;
```
which returns the error code, instead of the buff size.
I believe the translator (model) logic, needs some additional test, if either the parameters *or* the return value, is of "error" type then, issue generate support for it, otherwise, it needs some additional heuristic (hardcoded edge cases :)) to resolve the right footprint.